### PR TITLE
feat: Add no-unexpected-wire-adapter-usages linting rule

### PR DESCRIPTION
**Pull request by pmdartus**
## Changes

This PR introduces the `no-unexpected-wire-adapter-usages` rule. This rule offers a way to enforce that wire adapters are using along with `wire` decorator.

## Details

It's important to note that `no-unknown-wire-adapters` (#29) and this rule don't share the same configuration. Consumers will need to make sure to pass the same `adapters` list to both rules.

The name of this rule is really verbose, I am open to any other suggestion.

## Gus Work Item

W-7139651

### CODE REVIEW

**Comment by kevinv11n** [comment]
@jbleyleSF  can you please review all the copy?

File: docs/rules/no-unexpected-wire-adapter-usages.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+## Enforce wire adapters to be used with `wire` decorator (no-unexpected-wire-adapter-usages)
+
+This rule offers a way to enforce that wire adapters are using along with `wire` decorator. This prevents for example wire adapter: imperative invocation, reassignment, ...
```
Comment (kevinv11n) [inline docs/rules/no-unexpected-wire-adapter-usages.md]:
```suggestion
This rule offers a way to enforce that wire adapters are only used in the `wire` decorator. It prevents imperative invocation, reassignment, and so on.
```

File: docs/rules/no-unexpected-wire-adapter-usages.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+### Rule details
+
+This rule can be configured as followed:
```
Comment (kevinv11n) [inline docs/rules/no-unexpected-wire-adapter-usages.md]:
```suggestion
This rule can be configured as follows:
```

File: docs/rules/no-unexpected-wire-adapter-usages.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+## Enforce wire adapters to be used with `wire` decorator (no-unexpected-wire-adapter-usages)
+
+This rule offers a way to enforce that wire adapters are only used in the `wire` decorator. It prevents imperative invocation, reassignment, and so on.
```
Comment (ekashida) [inline docs/rules/no-unexpected-wire-adapter-usages.md]:
```suggestion
This rule prevents unexpected wire adapter usage by ensuring that they are only used via the `wire` decorator.
```

Comment (kevinv11n) [reply docs/rules/no-unexpected-wire-adapter-usages.md]: Great suggestion @ekashida

**Pull request merged**